### PR TITLE
Implement GetDistributedObjects [API-1916]

### PR DESCRIPTION
**Pull request by zpqrtbnk**
Implements `GetDistributedObjects` as is available in Java (see #799).

### CODE REVIEW

File: src/Hazelcast.Net/DistributedObjects/DistributedObjectFactory.cs [hunk 1]
```diff
@@ -57,6 +58,49 @@ public DistributedObjectFactory(Cluster cluster, SerializationService serializat
...
+                // at that point Java would actually make sure that the client has proxies for
+                // each object that was reported by the cluster - however that *cannot* work in
+                // .NET due to strong generics, because we cannot, for instance, create a
+                // HMap<TKey, TValue> object, since we don't know TKey nor TValue. this is not
+                // an issue really, proxies will be created on-demand.
+
+                result.Add(info); // build result
+            }
+
+            // locally destroy objects that are not remotely-known
```
Comment (emreyigit) [inline src/Hazelcast.Net/DistributedObjects/DistributedObjectFactory.cs:94]: I know it the logic is from Java but removing the local object is odd. Invoking a read only method modifies the client :S It should be idempotent, right? However, most probably, this is the reason that this method is not much promoted.

Comment (zpqrtbnk) [reply src/Hazelcast.Net/DistributedObjects/DistributedObjectFactory.cs:94]: agree, but I've tried to stay close to Java (for once)

File: src/Hazelcast.Net/IHazelcastClient.cs [hunk 2]
```diff
@@ -153,6 +153,12 @@ public interface IHazelcastClient : IAsyncDisposable
         /// </remarks>
         ValueTask DestroyAsync(IDistributedObject o);
 
+        /// <summary>
+        /// Gets information about all distributed objects known to the cluster.
+        /// </summary>
+        /// <returns>Information about all distributed objects know to the cluster.</returns>
+        Task<ICollection<DistributedObjectInfo>> GetDistributedObjectsAsync();
```
[the remark this answers is not in the document]
Comment (zpqrtbnk) [reply src/Hazelcast.Net/IHazelcastClient.cs]: makes sense, pushed a fix

**Review by emreyigit — APPROVED** [review]
Approved.

**Pull request merged**